Create notebook introducing VWB groups

emmarogge opened the pull request:
Create a notebook containing widgets to enable users to execute various VWB group management operations including:
- View groups in which the user is a member/admin
- View membership of groups in which user is an admin
- Create a new VWB group
- Delete a VWB group
- Add or remove a single user from VWB group
- Add or remove multiple users from VWB by providing a CSV file

emmarogge committed:
Create notebook with examples of VWB groups

Create a notebook containing widgets to enable users to execute various VWB group management operations including:
- View groups in which the user is a member/admin
- View membership of groups in which user is an admin
- Create a new VWB group
- Delete a VWB group
- Add or remove a single user from VWB group
- Add or remove multiple users from VWB by providing a CSV file

wnojopra commented on first_hour_on_terra/working_with_groups.ipynb:
```diff
@@ -0,0 +1,817 @@
+    "Perform common workspace resource operations including:\n",
+    "\n",
+    "1. Create a new Verily Workbench group.\n",
+    "1. Add and remove collaborators to your Verily Workbench group.\n",
```
Nit: "Add and remove collaborators to and from your Verily Workbench group" or alternatively, "Add/remove collaborators to/from your Verily Workbench group."

emmarogge replied:
fixed!

wnojopra commented on first_hour_on_terra/batch_template.csv:
```diff
@@ -0,0 +1,4 @@
+TERRA_USER_EMAIL,POLICY,
+amyu@google.com,MEMBER,
```
Lets put example emails here instead of actual emails. EXAMPLE@DOMAIN.com is fine, for example.

emmarogge replied:
thanks for catching -- fixed!

wnojopra reviewed (changes requested):
There's one gotcha that I've struggled with before when developing UI widgets in notebooks. Follow these instructions precisely to recreate a bug:
1) Execute the code that renders the "List users in group" UI.
2) Execute the code that renders the "Create group" UI.
3) Now go BACK to the "List users in group" UI. Attempt to list any group.
4) At this point you'll notice the "List users" UI doesn't get updated. BUT if you scroll down to the "Create group" UI, you'll see that its there instead!!

This is because the variable names in each cell are the same. When you executed number 2, you've overwritten the `output` variable. 

My suggestion to fix this would be to just make the variable names in each cell unique. `input_group_name` -> `list_input_group_name`. `output` -> `list_ouptut`, etc.

emmarogge committed:
Refactor introduction to include ref to docs, CLI

emmarogge committed:
Update for usability & reviewer feedback

- De-duplicate widget input field names for each widget.
- Create hyperlinks in Overview section to corresponding section/cell.
- Use placeholder emails in template CSV file.
- Rephrase awkward add/remove content in overview.

emmarogge committed:
Clarify instructions for add/remove user widget

emmarogge committed:
Refactor to encapsulate widgets for readability

Refactored each widget type as a class in order to maintain readability while preventing issues arising from variables with the same name. Moved the widget labels (with instructions) above the other fields so it looks reasonable after output is appended.

emmarogge committed:
Rephrase awkward instruction sentence

wnojopra commented on first_hour_on_terra/working_with_groups.ipynb:
```diff
@@ -0,0 +1,785 @@
+    "def users_table(x):\n",
+    "    \"\"\" Create table with data in a multiline\n",
+    "    string as first argument x)\"\"\"\n",
+    "    # html = f\"<table>\"\n",
```
Remove comment

emmarogge replied:
done!

wnojopra reviewed (approved):
Putting each widget in its own class ends up looking pretty clean. LGTM!

wnojopra commented on first_hour_on_terra/batch_template.csv:
```diff
@@ -0,0 +1,4 @@
+TERRA_USER_EMAIL,POLICY,
```
I don't think we should have commas at the end of each line, right? Because that indicates a third column who's values are just empty strings.

emmarogge replied:
fixed!

emmarogge committed:
Address reviewer feedback

Changes:
- Remove comment from users_table method.
- Remove empty column from template CSV.
